### DISCUSSION TITLE
console target for desktop apps

### DISCUSSION THREAD

**Discussion by ylazy** [airsdk/Adobe-Runtime-Support#1571] [Ideas]
Hi!

Sometimes I need an AIR Application without GUI, doesn't show up in the taskbar when running, able to write standard output, error level... and launch faster than applications with UI, is that possible?

Thanks!

**Comment by marchbold** [comment]
Hi,

Harman have added some initial support for this, currently the best support is done through running `adl` and passing your "app xml" to it and setting the `-cmd` option. eg:

```
adl -profile extendedDesktop -nodebug -cmd "app.xml" -- %*
```

You can see an example of this with [apm](https://github.com/airsdk/apm)

We have a basic actionscript class entry point (`APMConsoleApp.as`) that handles the INVOKE event. We compile that into `apm.swf` and run with a super basic app descriptor:

```xml
<?xml version="1.0" encoding="utf-8" ?>
<application xmlns="http://ns.adobe.com/air/application/33.1">
    <id>airsdk.apm</id>
    <filename>apm</filename>
    <name>apm</name>
    <versionNumber>0.0.1</versionNumber>
    <initialWindow>
        <content>apm.swf</content>
        <visible>false</visible>
    </initialWindow>
</application>
``` 

In terms of accessing stdout and stdin you can use:

```
System.output( "string" );
```

and 

```
var inputString:String = System.input();
```

from `flash.system.System`. 

Packaging it is a little weird currently. Windows you can produce an exe which seems to work, but on macOS it produces an app which you can call the contents directly but there's not really a command line output from AIR (as yet).

You can follow the development here:

https://github.com/airsdk/Adobe-Runtime-Support/issues/461
